Sources/StringFormat.swift

```diff
@@ -0,0 +1,92 @@
+import Foundation
+
+public enum StringFormat: RawRepresentable {
+    
+    public typealias RawValue = String
+    
+    /// Base64 encoded characters.
+    case byte
+    
+    /// Any sequence of octets.
+    case binary
+    
+    /// "full-date" representation, see RFC 3339, section 5.6.
+    case date
+    
+    /// "date-time" representation, see RFC 3339, section 5.6.
+    case dateTime
+    
+    /// Internet email address, see RFC 5322, section 3.4.1.
+    case email
+    
+    /// Internet host name, see RFC 1034, section 3.1.
+    case hostname
+    
+    /// IPv4 address, according to dotted-quad ABNF syntax as defined in RFC 2673, section 3.2.
+    case ipv4
+    
+    /// IPv6 address, as defined in RFC 2373, section 2.2.
+    case ipv6
+    
+    /// Used to hint UIs the input needs to be obscured.
+    case password
+    
+    /// A custom format
+    case other(String)
+    
+    /// A universal resource identifier (URI), according to RFC3986.
+    case uri
+    
+    public init(rawValue: RawValue) {
```

Comment:
Once you've typealiased RawValue = String, you can use String everywhere else in the type.